### PR TITLE
Space members invites : Part-3 of #949

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
This PR contains flow of space members invite as shown in reference video.

Reference Tickets :
https://github.com/acterglobal/a3/issues/949

Reference Balsamiq Design :
https://balsamiq.cloud/sj2u1jx/plw5law/r18F4

References Video:

https://github.com/acterglobal/a3/assets/51526480/cda42c95-c8d5-4405-811d-0f1468db6381

### CODE REVIEW

**Comment by kumarpalsinh25** [comment]
@gnunicorn 
Merging this PR so that we will have entire Invite flow on Thursday Release Build.

Still I would suggest to review this PR and raise separate issue if found any glitch or code quality issues so that I can resolve it on later PR.

File: app/lib/common/providers/space_providers.dart [hunk 1]
```diff
@@ -23,6 +23,29 @@ final spacesProvider =
...
+    .family<List<Space>, String>((ref, spaceId) async {
+  //GET LIST OF ALL SPACES
+  final allSpaces = ref.watch(spacesProvider);
+
+  //GET PARENT SPACE
+  final parentSpace = ref.watch(canonicalParentProvider(spaceId)).valueOrNull;
+  final parentSpaceId = parentSpace?.space.getRoomIdStr();
+
+  //GET LIST OF SPACES EXCLUDING PARENT SPACE
+  final spacesExcludingParentSpace = allSpaces
```
Comment (gnunicorn) [inline app/lib/common/providers/space_providers.dart:37]: Easier why not loop once and compare against both? Would safe time and memory...

File: app/lib/features/invite_members/pages/invite_space_members.dart [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      return;
+    }
+
+    EasyLoading.show(
+      status: L10n.of(context).invitingSpaceMembersLoading,
+      dismissOnTap: false,
+    );
+
+    try {
+      final currentSpace = ref.watch(spaceProvider(widget.roomId)).valueOrNull;
```
Comment (gnunicorn) [inline app/lib/features/invite_members/pages/invite_space_members.dart:163]: I know it's a bit confusing, but for one time call function, like actions or on-tap or something, these should be `ref.read` not `watch`...

File: app/lib/features/invite_members/pages/invite_space_members.dart [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            inviteCount++;
+          }
+        }
+      }
+      setState(() => selectedSpaces.clear());
+      if (!mounted) return;
+      EasyLoading.showToast(L10n.of(context).membersInvited(inviteCount));
+    } catch (e, st) {
+      _log.severe('Invite Space Members Error', e, st);
+      if (!mounted) return;
```
Comment (gnunicorn) [inline app/lib/features/invite_members/pages/invite_space_members.dart:187]: 👍

File: app/lib/features/invite_members/widgets/space_member_invite_card.dart [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    BuildContext context,
+    WidgetRef ref,
+    ProfileData profile,
+  ) {
+    final roomId = space.getRoomIdStr();
+    final title = profile.displayName?.isNotEmpty == true ? profile.displayName! : roomId;
+    final parent = ref.watch(canonicalParentProvider(roomId)).valueOrNull;
+    final parentRoomId = parent?.space.getRoomIdStr();
+
+    final avatar = ActerAvatar(
```
Comment (gnunicorn) [inline app/lib/features/invite_members/widgets/space_member_invite_card.dart:53]: Don't we already have a widget already doing all this?

**Review by gnunicorn — COMMENTED** [review]
Looks good. Just a few minor improvements

File: app/lib/common/providers/space_providers.dart (hunk 1 above)
Comment (kumarpalsinh25) [reply app/lib/common/providers/space_providers.dart:37]: True, let me do that

File: app/lib/features/invite_members/pages/invite_space_members.dart (hunk 2 above)
Comment (kumarpalsinh25) [reply app/lib/features/invite_members/pages/invite_space_members.dart:163]: Not confusing, it is my mistake. Need to `ref.read` on any function call for single time read operation where `ref. watch` need to use in widget tree for updating state.

Let me do needful changes here.

File: app/lib/features/invite_members/widgets/space_member_invite_card.dart (hunk 4 above)
Comment (kumarpalsinh25) [reply app/lib/features/invite_members/widgets/space_member_invite_card.dart:53]: Yes true but don't wanted to make existing widget to complex as this one contains trailing checkbox as well as members counts as sub-title. Event in future if we want to do any customisation with respect to space member invite design then we can use this separate card.

**Comment by kumarpalsinh25** [comment]
@gnunicorn 
I have address above feedbacks in below PR.
https://github.com/acterglobal/a3/pull/1778

**Pull request merged**